cdc/sink/ddlsink/mysql/async_ddl.go

```diff
@@ -16,24 +16,27 @@ package mysql
 import (
 	"context"
 	"database/sql"
+	"errors"
 	"fmt"
 	"time"
 
 	"github.com/pingcap/log"
 	timodel "github.com/pingcap/tidb/pkg/parser/model"
 	"github.com/pingcap/tiflow/cdc/model"
-	"github.com/pingcap/tiflow/pkg/errors"
 	"go.uber.org/zap"
 )
 
+// TODO: Use the flollowing SQL to check the ddl job status after tidb optimize
+// the information_schema.ddl_jobs table. Ref: https://github.com/pingcap/tidb/issues/55725
+//
+// SELECT JOB_ID, JOB_TYPE, SCHEMA_STATE, SCHEMA_ID, TABLE_ID, STATE, QUERY
+// FROM information_schema.ddl_jobs
 var checkRunningAddIndexSQL = `
-SELECT JOB_ID, JOB_TYPE, SCHEMA_STATE, SCHEMA_ID, TABLE_ID, STATE, QUERY
-FROM information_schema.ddl_jobs
+ADMIN SHOW DDL JOBS 1
```

Comment:
I'm not sure if this query will see an already finished DDL on the same table.

BTW DM uses `getDDLStatusFromTiDB` please see if we can unify it for our two projects.

Reply:
Here, TiCDC does not care about the finished ddl event, in fact, it only needs to ensure that there are no running add index ddl downstream of the same table before executing the next ddl event. In addition, the `getDDLStatusFromTiDB` function iterates through the job history to query for a particular ddl event, it may take more time than here.